### PR TITLE
Keep task add controls below cards

### PR DESCRIPTION
**Pull request by jonastemplestein**
## Summary

- keep the always-visible dashed add control immediately below the final task card in every row × status cell
- preserve empty-cell add controls and the cell-specific status/folder creation context
- replace render-time ref mutation and mirrored state with React 19 Effect Events and derived editor-path state
- add a DOM-order regression test for populated and empty cells

## Root cause

PR #1985 followed a contradictory earlier instruction and moved the add controls back above each list, undoing #1982. The clarified behavior is below the last card in each individual cell.

## Validation

- React Doctor: 100/100, 0 diagnostics for changed files
- `pnpm typecheck`
- `pnpm lint` (0 warnings, 0 errors)
- `pnpm test` (OS: 1,407 passed, 2 skipped; all workspace suites green)
- focused task suites: 23 passed
- browser-verified desktop and 390×844 mobile layouts
- browser-verified cell context, selected initial heading, automatic path slugging, and Cmd+Return creation

---

> [!NOTE]
> **Low Risk**
> Scoped repo task board UI and local editor state; no auth, persistence APIs, or security-sensitive paths.
> 
> **Overview**
> Moves each board cell’s dashed **Add task** control from above the card list to **immediately below** the last task card (empty cells still show the button). **`TaskColumn`** is exported so layout can be asserted in tests.
> 
> Task selection during renames no longer mirrors URL in separate `editorPath` state: a short-lived **`editorPathOverride`** (`source` → `target`) drives the open editor until `selectedPath` catches up, via **`reconcileEditorPathOverride`**.
> 
> Draft persistence on unmount and the **`c`** keyboard shortcut use React 19 **`useEffectEvent`** instead of render-time ref updates; the sheet’s “path was edited” flag is a ref. Delete only opens the confirm dialog when **`resolvePath()`** succeeds.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c27faff4894b50c73b5ef5eefec04c1c4148114a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

| Group | Lines | Significant |
| --- | ---: | ---: |
| UI components | +70 -58 | +66 -58 🟩🟩🟩🟥🟥 |
| Tests | +83 -0 | +74 -0 🟩🟩🟩⬜⬜ |
| **Total** | **+153 -58** | **+140 -58** 🟩🟩🟩🟩🟥 |

<sub>Lines counts every changed line; Significant ignores blank lines and JS comments. Between 985e5d1 and c27faff, bucketed first-match-wins into the groups defined in `scripts/ci/loc-report.ts`.</sub>

## Environment Config Lease

<details>
<summary>Slot: preview-3 | Doppler config: preview_3</summary>

| app | status | commit | preview | size (gzip) | deploy duration | test duration | retries | cleanup duration | workflow run | updated | summary |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Auth | cleanup failed | `c27faff` | [https://auth.iterate-preview-3.com](https://auth.iterate-preview-3.com) | 819.7 KiB | 18.6s | 206ms |  | 1.7s | [Workflow run](https://github.com/iterate/iterate/actions/runs/502549181171889) | 2026-07-14T14:28:54.793Z | CloudflareApiError: Cloudflare API POST /accounts/376ef7ed81b0573f93524de763666c15/d1/database/2017a0b9-ebd5-482c-b2b8-894a27733dbc/query failed (429): [{"code":971,"message":"Ple... |
| OS | released | `c27faff` | [https://os.iterate-preview-3.com](https://os.iterate-preview-3.com) | 4.31 MiB (+0.0 KiB vs main) | 85.2s | 222.0s | 3 retried: agent answers after llm model is selected (vitest x1) · DESIRED: a live-capability fetch handler serves WebSockets at an app host (vitest x1) · hydrates the client-only integrations route without rebuilding the shell (specs x1) | 12.5s | [Workflow run](https://github.com/iterate/iterate/actions/runs/502549181171889) | 2026-07-14T14:28:51.823Z | Preview app released. |
| Semaphore | released | `c27faff` | [https://semaphore.iterate-preview-3.com](https://semaphore.iterate-preview-3.com) | 440.8 KiB | 14.6s | 6.3s |  | 496ms | [Workflow run](https://github.com/iterate/iterate/actions/runs/502549181171889) | 2026-07-14T14:28:39.798Z | Preview app released. |
| Streams Example App | released | `c27faff` | [https://streams.iterate-preview-3.com](https://streams.iterate-preview-3.com) | 1.43 MiB | 17.4s | 20.2s |  | 731ms | [Workflow run](https://github.com/iterate/iterate/actions/runs/502549181171889) | 2026-07-14T14:28:53.810Z | Preview app released. |

</details>

<details>
<summary>Auth failure details</summary>

<pre>CloudflareApiError: Cloudflare API POST /accounts/376ef7ed81b0573f93524de763666c15/d1/database/2017a0b9-ebd5-482c-b2b8-894a27733dbc/query failed (429): [{"code":971,"message":"Please wait and consider throttling your request speed"}]
    at cfV4 (/home/runner/work/iterate/iterate/scripts/lib/env-context.ts:108:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
    at async wipeD1Tables (/home/runner/work/iterate/iterate/scripts/lib/deploy-helpers.ts:312:5)
    at async eraseData (/home/runner/work/iterate/iterate/apps/auth/scripts/erase-data.ts:44:3)
    at async Command.&lt;anonymous&gt; (/home/runner/work/iterate/iterate/node_modules/.pnpm/trpc-cli@0.15.1_@orpc+server@1.13.14_@opentelemetry+api@1.9.0_crossws@0.4.4_srvx@0.11.1_4eb263ef2d44bd3d623d178b0159b94d/node_modules/trpc-cli/dist/index.js:416:32)
    at async Command.parseAsync (/home/runner/work/iterate/iterate/node_modules/.pnpm/commander@14.0.3/node_modules/commander/lib/command.js:1122:5)
    at async Command.&lt;anonymous&gt; (/home/runner/work/iterate/iterate/node_modules/.pnpm/trpc-cli@0.15.1_@orpc+server@1.13.14_@opentelemetry+api@1.9.0_crossws@0.4.4_srvx@0.11.1_4eb263ef2d44bd3d623d178b0159b94d/node_modules/trpc-cli/dist/index.js:457:21)
    at async Command.parseAsync (/home/runner/work/iterate/iterate/node_modules/.pnpm/commander@14.0.3/node_modules/commander/lib/command.js:1122:5)
    at async Object.run (/home/runner/work/iterate/iterate/node_modules/.pnpm/trpc-cli@0.15.1_@orpc+server@1.13.14_@opentelemetry+api@1.9.0_crossws@0.4.4_srvx@0.11.1_4eb263ef2d44bd3d623d178b0159b94d/node_modules/trpc-cli/dist/index.js:527:9) {
  method: 'POST',
  path: '/accounts/376ef7ed81b0573f93524de763666c15/d1/database/2017a0b9-ebd5-482c-b2b8-894a27733dbc/query',
  status: 429
}

&gt; @iterate-com/auth@0.0.0-dev destroy /home/runner/work/iterate/iterate/apps/auth
&gt; tsx ./scripts/erase-data.ts --env preview_3

Erasing all data in preview_3 (auth D1 2017a0b9-ebd5-482c-b2b8-894a27733dbc)
 ELIFECYCLE  Command failed with exit code 1.</pre>

</details>

### CODE REVIEW

File: apps/os/src/components/repo-ide/repo-tasks-view.tsx [hunk 1]
```diff
@@ -907,7 +907,7 @@ function TaskEditorSheet({
                     className="shrink-0 text-muted-foreground hover:text-destructive"
                     title="Delete task"
                     aria-label="Delete task"
-                    onClick={() => withResolvedPath(() => setDeleteOpen(true))}
+                    onClick={openDeleteDialog}
```
[the remark this answers is not in the document]
Comment (jonastemplestein) [reply apps/os/src/components/repo-ide/repo-tasks-view.tsx]: Addressed in c27faff48: the one-use delete wrapper is inlined on the button.

File: apps/os/src/components/repo-ide/repo-tasks-view.tsx [hunk 2]
```diff
@@ -119,11 +119,14 @@ export function RepoTasksView({
...
-  draftRef.current = draft;
-  onSetWorkingRef.current = onSetWorking;
+  const [editorPathOverride, setEditorPathOverride] = useState<{
+    source: string | undefined;
+    target: string | undefined;
+  }>();
+  const editorPath =
+    editorPathOverride !== undefined && editorPathOverride.source === selectedPath
+      ? editorPathOverride.target
+      : selectedPath;
```
[the remark this answers is not in the document]
Comment (jonastemplestein) [reply apps/os/src/components/repo-ide/repo-tasks-view.tsx:127]: Fixed in c27faff48. Pending editor-path overrides now expire as soon as the router selection advances, so a later navigation back to the source path cannot reopen the stale target. Added a regression test covering source → target → source.
